lib/redis-session-store.rb

```diff
@@ -97,11 +108,13 @@ def find_session(req, sid)
 
   def load_session_from_redis(redis_connection, req, sid)
     return nil unless sid
-    if read_fallback
-      data = redis_connection.get(prefixed(sid)) || redis_connection.get(prefixed_fallback(sid))
-    else
-      data = redis_connection.get(prefixed(sid))
-    end
+    data = if read_primary && !read_fallback
+             redis_connection.get(prefixed(sid))
+           elsif read_primary && read_fallback
+             redis_connection.get(prefixed(sid)) || redis_connection.get(prefixed_fallback(sid))
+           elsif !read_primary && read_fallback
+             redis_connection.get(prefixed_fallback(sid))
+           end
```

Comment:
is this intended to be long-term code?

my first thought when looking at it was it's missing a case (2 booleans, 4 total branches), I know that we check/prevent the missing 4th case in the initializer, but maybe if we swapped these booleans for an enum, it might be easier? then we have two configs instead of 4?

```
session_id_read_mode: one of [:private_only, :private_with_public_fallback, :public_only]
session_id_write_mode: (same list)
```

Reply:
I wasn't super happy with what I have now, I like that a lot more, will update!